src/features/impl_rkyv.rs

```diff
@@ -166,27 +167,36 @@ mod u64 {
 
 #[cfg(test)]
 mod test {
-    pub type DefaultSerializer = rkyv::ser::serializers::CoreSerializer<256, 256>;
-    pub type DefaultDeserializer = rkyv::Infallible;
-    use rkyv::ser::Serializer;
-    use rkyv::*;
+    /// The serializer type expected by [`rkyv::to_bytes()`].
+    pub type TestSerializer<'a> = rkyv::api::high::HighSerializer<
+        rkyv::util::AlignedVec,
+        rkyv::ser::allocator::ArenaHandle<'a>,
+        rkyv::rancor::Panic,
+    >;
+    /// The deserializer type expected by [`rkyv::deserialize()`].
+    pub type TestDeserializer = rkyv::api::high::HighDeserializer<rkyv::rancor::Panic>;
+    pub type TestValidator<'a> = rkyv::api::high::HighValidator<'a, rkyv::rancor::Panic>;
     pub fn test_archive<T>(value: &T)
     where
-        T: core::fmt::Debug + PartialEq + rkyv::Serialize<DefaultSerializer>,
-        T::Archived: core::fmt::Debug + PartialEq<T> + rkyv::Deserialize<T, DefaultDeserializer>,
+        T: core::fmt::Debug
+            + PartialEq
+            + rkyv::Portable
+            + for<'a> rkyv::Serialize<TestSerializer<'a>>,
+        T::Archived: core::fmt::Debug + PartialEq<T> + rkyv::Deserialize<T, TestDeserializer>,
+        // #[cfg(feature = "bytecheck")]
+        T::Archived: for<'a> rkyv::bytecheck::CheckBytes<TestValidator<'a>>,
```

Comment:
I don't think we can test `access()` and must always use `access_unchecked()`, because `cfg`'s don't work in `where` clauses and `access()` mandates `bytecheck` support.